src/lib.rs

```diff
@@ -73,7 +71,7 @@ mod tls;
 pub use tls::TlsStream;
 
 pub use crate::consumer::{Consumer, ConsumerBuilder};
-pub use crate::error::FaktoryError;
+pub use crate::error::{Error, ProtocolError};
```

Comment:
I'm tempted to only `pub use Error`, and then make `mod error` `pub` for access to "deeper" error types.

Reply:
If we do that, I think we should also remove `Error` from `ProtocolError` to avoid stuttering (`error::ProtocolError` -> `error::Protocol`). This is a topic under ongoing debate (https://github.com/rust-lang/api-guidelines/discussions/66), but I think that's the way things are generally leaning.

Reply:
Big fan of this!